core/src/main/scala/neotypes/Driver.scala

```diff
@@ -8,14 +8,42 @@ import org.neo4j.driver.v1.{AccessMode, Driver => NDriver}
 import scala.jdk.CollectionConverters._
 import scala.language.higherKinds
 
+/** neotypes driver for accessing a neo4j graph database
+  *
+  * @param driver neo4j driver
+  * @tparam F effect type for driver
+  */
 final class Driver[F[_]](private val driver: NDriver) extends AnyVal {
+
+  /** acquire session to a database in read mode
+    * @note a session is a resource that needs to be allocated and released
+    *
+    * @param F asynchronous effect type with resource type defined
+    * @tparam R effect type that allows creating, using, and releasing a resource
+    * @return Session[F] in effect type R
+    */
   def session[R[_]](implicit F: Async.Aux[F, R]): R[Session[F]] =
     session[R](accessMode = AccessMode.READ)
 
+  /** acquire session to a database in read or write mode
+    * @note a session is a resource that needs to be allocated and released
+    *
+    * @param accessMode read or write mode
+    * @param bookmarks bookmarks passed between transactions for neo4j casual chaining
+    * @param F asynchronous effect type with resource type defined
+    * @tparam R effect type that allows creating, using, and releasing a resource
+    * @return Session[F] in effect type R
+    */
   def session[R[_]](accessMode: AccessMode, bookmarks: String*)
                    (implicit F: Async.Aux[F, R]): R[Session[F]] =
     F.resource(createSession(accessMode, bookmarks))(session => session.close)
 
+  /** create a session from the neo4j driver
```

Comment:
This is a private method, it is not needed to be documented as it won't be visible.